src/operators/lazysum.jl

```diff
@@ -0,0 +1,76 @@
+"""
+    LazySum{O} <: AbstractVector{O}
+
+Type that represents a lazy sum i.e explicit summation is only done when needed. 
+This type is basically an AbstractVector with some extra functionality to calcaulate things efficiently.
+
+## Fields
+- ops -- Vector of summable objects
+
+---
+
+## Constructors
+    LazySum(x::Vector)
+
+"""
+struct LazySum{O} <: AbstractVector{O}
+    ops::Vector{O}
+
+end
+
+Base.size(x::LazySum) = size(x.ops)
+Base.getindex(x::LazySum, i) = x.ops[i]
+#iteration and summation gets automatically implementend thanks to subtyping
+
+Base.length(x::LazySum) = prod(size(x))
+
+# Holy traits
+struct TimeDependent end
+struct NotTimeDependent end
+
+TimeDependence(x) = NotTimeDependent()
+TimeDependence(x::LazySum) = istimed(x) ? TimeDependent() : NotTimeDependent()
+
+istimed(::TimeDependent) = true
+istimed(::NotTimeDependent) = false
+istimed(x) = istimed(TimeDependence(x))
+istimed(x::LazySum) = any(istimed, x)
+
+# constructors
+LazySum(x) = LazySum([x])
+LazySum(f::Function,x) = LazySum(map(y->f(y),x))
+
+# Internal use only, works always
+_eval_at(x,args...) = x # -> this is what you should define for your custom structs inside a LazySum
+
+# wrapper around _eval_at
+eval_at(x,args...) = eval_at(TimeDependence(x),x,args...)
+eval_at(::TimeDependent,x::LazySum,t::Number) = LazySum(O -> _eval_at(O,t),x) 
+eval_at(::TimeDependent,x::LazySum) = throw(ArgumentError("attempting to evaluate time-dependent LazySum without specifiying a time"))
+eval_at(::NotTimeDependent,x::LazySum) = sum(O -> _eval_at(O),x)
+eval_at(::NotTimeDependent,x::LazySum,t::Number) = throw(ArgumentError("attempting to evaluate time-independent LazySum at time"))
+
+# For users
+(x::LazySum)() = eval_at(x)
+ConvertOperator(x::LazySum) = eval_at(x) # using ConvertOperator should do explicit multiplication
+(x::LazySum)(t::Number) = eval_at(x,t) # using (t) should return NotTimeDependent LazySum
+ConvertOperator(x::LazySum,t) = x(t)() 
```

Comment:
I am not a big fan of having `lazysum()` automatically sum up the elements. This does not really feel like intuitive syntax. I also think the capitalized `ConvertOperator` is misleading, this seems to suggest that it is a special type you are constructing. Is it an option to just replace this with `Base.sum`?